File: src/test/java/cspom/variable/BooleanTest.java
package cspom.variable;

import static org.junit.Assert.*;

import java.util.Arrays;

import org.junit.Test;

public final class BooleanTest {

    @Test
    public void testValueOf() {
        assertSame(BooleanDomain.TRUE, BooleanDomain.valueOf(true));
        assertSame(BooleanDomain.FALSE, BooleanDomain.valueOf(false));
    }

    @Test
    public void testIsConstant() {
        assertTrue(BooleanDomain.TRUE.isConstant());
        assertTrue(BooleanDomain.FALSE.isConstant());
        assertFalse(BooleanDomain.DOMAIN.isConstant());
    }

    @Test
    public void testToString() {
        assertEquals("true", BooleanDomain.TRUE.toString());
    }

    @Test
    public void testGetBoolean() {
        assertSame(BooleanDomain.TRUE.getBoolean(), true);
        assertSame(BooleanDomain.FALSE.getBoolean(), false);
    }

    @Test(expected = IllegalStateException.class)
    public void testGetBooleanDomain() {
        BooleanDomain.DOMAIN.getBoolean();
    }

    @Test
    public void testGetValues() {
        assertEquals(BooleanDomain.DOMAIN.getValues(), Arrays.asList(false, true));
    }

}

File: src/test/scala/cspom/variable/BooleanTest.scala
package cspom.variable
import org.junit.Assert._
import org.junit.Test
final class BooleanTest {

  @Test
  def testValueOf() {
    assertSame(TrueDomain, BooleanDomain.valueOf(true));
    assertSame(FalseDomain, BooleanDomain.valueOf(false));
  }

  @Test
  def testIsConstant() {
    assertTrue(TrueDomain.isConstant);
    assertTrue(FalseDomain.isConstant);
    assertFalse(UnknownBooleanDomain.isConstant);
  }

  @Test
  def testToString() {
    assertEquals("true", TrueDomain.toString());
    assertEquals("false", FalseDomain.toString());
  }

  @Test
  def testGetBoolean() {
    assertSame(TrueDomain.getBoolean, true);
    assertSame(FalseDomain.getBoolean, false);
  }

  @Test(expected = classOf[UnsupportedOperationException])
  def testGetBooleanDomain() {
    UnknownBooleanDomain.getBoolean;
  }

  @Test
  def testGetValues() {
    assertEquals(UnknownBooleanDomain.getValues, List(false, true));
  }

}